simulation/app.py

```diff
@@ -0,0 +1,621 @@
+    percent['condition'].append('infected')
+    percent['percent_value'].append(data['groups']['infected'])
+    percent['condition'].append('recovered')
+    percent['percent_value'].append(data['groups']['recovered'])
+    percent['condition'].append('dead')
+    percent['percent_value'].append(data['groups']['dead'])
+
+    df = pd.DataFrame(data=percent)
+
+    fig = px.pie(df, values='percent_value', names='condition', color='condition',
+                 color_discrete_map={
+                        'susceptible': 'blue',
+                        'infected': 'red',
+                        'recovered': 'green',
+                        'dead': 'gray'
+                 })
+    fig.update_traces()
+    fig.update_layout({
+        'title': {'text': 'Percent of population in each group', 'font': {'size': 30}},
+        'template': 'plotly_dark'
+    })
+    return fig
+
+
+@app.callback(Output('current-population-amount', 'children'),
+              [Input('interval-component', 'n_intervals'),
+               Input('simulation-data', 'data')])
+def update_population_amount(n_intervals: int, json_data) -> 'html table':
+    """
+    Function handling display html table with amount of each group.
+
+    Parameters
+    ----------
+    n_intervals: int, required
+        Number of past intervals.
+    json_data: json, required
+        JSON contain data of simulation.
+
+    Returns
+    -------
+    Html table base od bash html.Table object.
+    """
+
+    data = json.loads(json_data)
+    return [html.Thead(html.Tr([html.Td('Condition'), html.Td('Amount')])),
+            html.Tr([html.Td('Susceptible'), html.Td(data['groups']['susceptible'])]),
+            html.Tr([html.Td('Infected'), html.Td(data['groups']['infected'])]),
+            html.Tr([html.Td('Recovered'), html.Td(data['groups']['recovered'])]),
+            html.Tr([html.Td('Dead'), html.Td(data['groups']['dead'])])]
+
+
+@app.callback(Output('additional-parameters', 'children'),
+              [Input('interval-component', 'n_intervals'),
+              Input('simulation-data', 'data')],
+              [State('susceptible-amount', 'value')])
+def parameter_update(n_intervals: int, json_data: json, susceptible_amount: int) -> 'html table':
+    """
+    Function handling display html table with additional parameters q and R0.
+
+    Parameters
+    ----------
+    n_intervals: int, required
+        Number of past intervals.
+    json_data: json, required
+        JSON contain data of simulation.
+    susceptible_amount: int, required
+        Start number of susceptible.
+
+    Returns
+    -------
+    Html table base od bash html.Table object with parameter q and R0.
+    """
+
+    data = json.loads(json_data)
+    change_susceptible = data['groups']['susceptible'] - old_amounts['susceptible']
+    change_d = abs(old_amounts['dead'] - data['groups']['dead'])
+    change_r = abs(old_amounts['recovered'] - data['groups']['recovered'])
+    change_removed = change_d + change_r
```

Comment:
Maybe better to place such functionality in separate functions? To somehow segragate, for better visibility of functionality.